src/specVersions/v0.0/RicardianContractProcessorImpl.ts

```diff
@@ -62,19 +59,40 @@ export class RicardianContractProcessorImpl implements RicardianContractProcesso
     this.wrappedHelpers.push(name)
   }
 
+  protected registerHelper(name: string, fn: Handlebars.HelperDelegate) {
+    Handlebars.registerHelper(name, fn)
+  }
+
   public getSpecVersion() {
     return {
       major: 0,
       minor: 0,
     }
   }
 
+  /**
+   * Process the RicardianContractConfig and return a RicardianContract.
+   *
+   * @param config A `RicardianContractConfig` object
+   */
   /**
    * Process the RicardianContractConfig and return a RicardianContract.
    *
    * @param config A `RicardianContractConfig` object
    */
   public process(config: RicardianContractConfig): RicardianContract {
+    const  version = getContractSpecVersion(config)
+    const specVersion = this.getSpecVersion()
```

Comment:
do you see this validate the version logic being used in each subsequent `RicardianContractProcessor` version? if so maybe in a future refactor we may just to implement this in a base abstract class. Unless you think all future version will also extend this v0.0 implementation || extend the previous version like in v0.1: `RicardianContractProcessorImpl extends RCP_0_0`

Reply:
Yeah, a refactor is definitely needed here. I keep kind of fighting this thing and still haven't hit on a code organization I really like. As for future versions it's hard to say. I figure for purely additive changes (like adding new helpers) each can just keep building on the prior version. Tho that may make for a rather long and ugly inheritance chain.

Reply:
Yeah the inheritance chain is what i was worried about as well. Maybe we can discuss / get some feedback and decide how to tackle this